components/website/utils/UnderlineAnimation/index.jsx

```diff
@@ -1,28 +1,72 @@
-import {useState } from 'react';
-import Fade from "react-reveal/Fade";
+import { useState, useEffect } from 'react';
 
 import Card from "/components/utils/Card";
 
 import styles from './style.module.css';
 
-export default function UnderlineAnimation({children, text}) {
-    const [hovered, setHover] = useState(false);
+export default function UnderlineAnimation({children, text, afterText}) {
+    const extendedMargin = -220;
+    const retractedMargin = -20;
+    const speed = 4;
+    const fps = 60;
+
+    const defaultState = {
+        status: 0,
+        margin: extendedMargin,
+    };
+    
+    const [st, updateState] = useState(defaultState);
+
+    const scrollDown = () => {
+        if(st.margin == retractedMargin) {
+            updateState({
+                status: 0,
+                margin: retractedMargin,
+            });
+        } else {
+            updateState({
+                status: st.status,
+                margin: st.margin + speed,
+            });
+        }
+    }
+
+    const scrollUp = () => {
+        if(st.margin == extendedMargin) {
+            updateState({
+                status: 0,
+                margin: extendedMargin,
+            });
+        } else {
+            updateState({
+                status: st.status,
+                margin: st.margin - speed,
+            });
+        }
+    }
+
+    useEffect(() => {
+        if(st.status == 1) {
+            setTimeout(scrollDown, 1000 / fps);
+        } else if(st.status == 3) {
+            setTimeout(scrollUp, 1000 / fps);
+        }        
+      }, [st]);
+
     return (
-        <span className="relative leading-none z-10 w-auto my-4 inline-block">
-                <h2 className="font-bold text-white md:w-full relative z-0 border-b-8 border-b-solid border-white">
-                    <span className="relative h-auto inline-block z-0" onMouseEnter={() => setHover(true)} 
-                        onMouseLeave={() => setHover(false)}>
-                            {children}
-                    </span>
-                </h2>
-
-                <div className={styles.cardWrapper}>
-                    <Fade top when={hovered} distance="10px">
-                        <Card className="z-0" img="/images/mascot-footer.svg" alt="MascotFooter" inverted={true}>
-                            <h5 className={`font-iregular font-normal ${styles.cardText}`}>{text}</h5>
-                        </Card>
-                    </Fade> 
-                </div>
+        <span className="relative leading-none z-10 w-full inline-block z-20 mt-4">
+            <div className="absolute bottom-0 w-full h-52 bg-secondary z-20" style={{width: "500px"}}></div>
+            <span className="relative h-full inline-block z-20 border-b-8 border-white" onMouseEnter={() => updateState({status:1, margin:st.margin})} 
```

Comment:
```suggestion
            <span className="relative h-full inline-block z-20 border-b-8 border-white" onMouseEnter={() => updateState((oldState) => {...oldState, status: 1})} 
```

When updating state based on the previous state, pass an update function to updateState in order to be consistent with concurrent updates. Since you are only updating one field and keeping everything as it was, you can make it even cleaner with the `...` operator.